components/org.wso2.bfsi.identity.extensions/src/main/java/org/wso2/bfsi/identity/extensions/auth/extensions/response/handler/impl/BFSIDefaultResponseTypeHandlerImpl.java

```diff
@@ -0,0 +1,171 @@
+                String consentScope = consentIdClaim + consentID;
+                if (!Arrays.asList(scopes).contains(consentScope)) {
+                    String[] updatedScopes = ArrayUtils.addAll(scopes, consentScope);
+                    if (log.isDebugEnabled()) {
+                        log.debug(String.format("Updated scopes: %s", Arrays.toString(updatedScopes)
+                                .replaceAll("[\r\n]", "")));
+                    }
+                    return updatedScopes;
+                }
+            }
+
+        } else {
+            return new String[0];
+        }
+
+        return oAuthAuthzReqMessageContext.getApprovedScope();
+    }
+
+    /**
+     * Call sessionDataAPI and retrieve request object, decode it and return consentID.
+     *
+     * @param sessionDataKey sessionDataKeyConsent parameter from authorize request
+     * @return consentID
+     */
+    String getConsentIDFromSessionData(String sessionDataKey) {
+
+        String consentID = StringUtils.EMPTY;
+        if (sessionDataKey != null && !sessionDataKey.isEmpty()) {
+            RequestObjectService requestObjectService = IdentityExtensionsDataHolder.getInstance()
+                    .getRequestObjectService();
+            if (requestObjectService != null) {
+                consentID = retrieveConsentIDFromReqObjService(requestObjectService, sessionDataKey);
+                if (consentID.isEmpty()) {
+                    log.warn("Failed to retrieve ConsentID from query parameters");
+                }
+            } else {
+                log.warn("Failed to retrieve Request Object Service");
+            }
+        } else {
+            log.warn("Invalid Session Data Key");
+        }
+        return consentID;
+    }
+
+    /**
+     * Call Request Object Service and retrieve consent id.
+     *
+     * @param service        request object service
+     * @param sessionDataKey session data key
+     * @return consentID
+     */
+    String retrieveConsentIDFromReqObjService(RequestObjectService service, String sessionDataKey) {
+
+        String consentID = StringUtils.EMPTY;
+        try {
+            List<RequestedClaim> requestedClaims = service.getRequestedClaimsForSessionDataKey(sessionDataKey,
+                    false);
+            consentID = iterateClaims(requestedClaims);
+            if (consentID.isEmpty()) {
+                requestedClaims = service.getRequestedClaimsForSessionDataKey(sessionDataKey, true);
+                consentID = iterateClaims(requestedClaims);
+            }
+
+        } catch (RequestObjectException ex) {
+            log.warn("Exception occurred", ex);
+        }
+        return consentID;
+    }
+
+    /**
+     * Iterate the claims list to identify the consent-ID.
+     *
+     * @param requestedClaims list of claims
+     * @return consent id
+     */
+    String iterateClaims(List<RequestedClaim> requestedClaims) {
```

Comment:
Shall we use a better method name since this method actually returns the consent id?

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/6/commits/fc5b5f2d081c4d52de5072aa80c4cafa18e006b3